middleware/middleware18_test.go

```diff
@@ -0,0 +1,71 @@
+// +build go1.8
+
+package middleware
+
+import (
+	"crypto/tls"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+
+	"golang.org/x/net/http2"
+)
+
+// NOTE: we must import `golang.org/x/net/http2` in order to explicitly enable
+// http2 transports for certain tests. The runtime pkg does not have this dependency
+// though as the transport configuration happens under the hood on go 1.7+.
+
+func TestWrapWriterHTTP2(t *testing.T) {
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, cn := w.(http.CloseNotifier)
+		if !cn {
+			t.Fatal("request should have been a http.CloseNotifier")
+		}
+		_, fl := w.(http.Flusher)
+		if !fl {
+			t.Fatal("request should have been a http.Flusher")
+		}
+		_, hj := w.(http.Hijacker)
+		if hj {
+			t.Fatal("request should not have been a http.Hijacker")
+		}
+		_, rf := w.(io.ReaderFrom)
+		if rf {
+			t.Fatal("request should not have been a io.ReaderFrom")
+		}
+		_, ps := w.(http.Pusher)
+		if !ps {
+			t.Fatal("request should have been a http.Pusher")
+		}
+
+		w.Write([]byte("OK"))
+	})
+
+	server := http.Server{
+		Addr:    ":7072",
+		Handler: handler,
```

Comment:
This handler should be wrapped in a middleware to have this test check that the response writer still has the interfaces expected after wrapping.